web-app/react-app/components/ProfilePage/IdentityHandler/index.jsx

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import React, {Component} from 'react'
+import {Panel, Row, Alert} from 'react-bootstrap'
+import IntegrationKeyHandlerSegment from '../IntegrationKeyHandler/IntegrationKeyHandlerSegment'
+import type {IntegrationKey} from '../../../flowtype/integration-key-types'
+import {createIdentity} from '../../../actions/integrationKey'
+import {connect} from 'react-redux'
+import type {ErrorInUi} from '../../../flowtype/common-types'
+import type {IntegrationKeyState} from '../../../flowtype/states/integration-key-state'
+import {deleteIntegrationKey, getIntegrationKeysByService} from '../../../actions/integrationKey'
+import getWeb3 from '../../../utils/web3Provider'
+
+type StateProps = {
+    error: ?ErrorInUi,
+    integrationKeys: Array<IntegrationKey>
+}
+
+type DispatchProps = {
+    createIdentity: (integrationKey: IntegrationKey) => void,
+    deleteIntegrationKey: (id: $ElementType<IntegrationKey, 'id'>) => void,
+    getIntegrationKeysByService: (service: $ElementType<IntegrationKey, 'service'>) => void
+}
+
+type Props = StateProps & DispatchProps
+
+const service = 'ETHEREUM_ID'
+
+export class IdentityHandler extends Component<Props> {
+    componentDidMount() {
+        // TODO: Move to (yet non-existent) router
+        this.props.getIntegrationKeysByService(service)
+    }
+
+    onNew = (integrationKey: IntegrationKey) => {
+        const name = integrationKey.name
+        delete integrationKey.name
+        this.props.createIdentity({
+            name,
+            service,
+            json: integrationKey,
+        })
+    }
+
+    onDelete = (id: $ElementType<IntegrationKey, 'id'>) => {
+        this.props.deleteIntegrationKey(id)
+    }
+
+    render() {
+        const hasWeb3 = getWeb3().isEnabled()
+        return (
+            <Panel header="Ethereum Identities">
+                <p key={0}>
```

Comment:
Why do we `key` it? It's not an array, is it?

Reply:
Oh, my bad. It was an array for a (very little) while and forgot them...